### PR TITLE
[BPK-1133] Add theming to datepickers

### PR DESCRIPTION
**Pull request by matthewdavidson**
Ronseal.

### CODE REVIEW

File: packages/bpk-component-datepicker/src/themeAttributes.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import { themeAttributes as calendarAttributes } from 'bpk-component-calendar';
+import { themeAttributes as modalAttributes } from 'bpk-component-modal';
+import { themeAttributes as popoverAttributes } from 'bpk-component-popover';
+
+export default [
+  ...calendarAttributes,
+  ...popoverAttributes,
+  ...modalAttributes,
+].filter((attribute, index, attributes) => attributes.indexOf(attribute) === index);
```
Comment (matteo-hertel) [inline packages/bpk-component-datepicker/src/themeAttributes.js]: I'm just curious here, what this filter is supposed to catch? `attributes.indexOf(attribute) === index` will always evaluate to `true` right?

Comment (matthewdavidson) [reply packages/bpk-component-datepicker/src/themeAttributes.js]: This is a `uniq` filter. Any items that match but are not at the same index are tossed.

Comment (matthewdavidson) [reply packages/bpk-component-datepicker/src/themeAttributes.js]: In this case, modal and popover have similar (if not identical) theme attributes .. this accounts for any instance in which their theme attributes might change.

File: packages/bpk-component-datepicker/stories.js [hunk 2]
```diff
@@ -57,15 +57,21 @@ class CalendarContainer extends Component {
...
+        <div id="application-element">
+          <BpkDatepicker
+            {...this.props}
+            date={this.state.date}
+            onDateSelect={(date) => {
+              this.setState({ date });
+              action('Selected date')(date);
+            }}
+            onMonthChange={action('Changed month')}
+            getApplicationElement={() => document.getElementById('application-element')}
```
Comment (k0nserv) [inline packages/bpk-component-datepicker/stories.js]: nit: Shouldn't we prefer the ref method instead?

File: packages/bpk-component-datepicker/src/themeAttributes.js (hunk 1 above)
Comment (matteo-hertel) [reply packages/bpk-component-datepicker/src/themeAttributes.js]: ah! nice!

File: packages/bpk-component-datepicker/stories.js (hunk 2 above)
Comment (matthewdavidson) [reply packages/bpk-component-datepicker/stories.js]: Could do, but a lot of usage in Skyscanner (particularly with OC components) depends on containers defined outwith the components react tree (e.g. the header culture selector needs to get application element defined in microsite, not in header).

Comment (k0nserv) [reply packages/bpk-component-datepicker/stories.js]: Good point, but shouldn't we use the React-way when we can?

Comment (matthewdavidson) [reply packages/bpk-component-datepicker/stories.js]: Myself and hugo caught up offline and discussed that the most common usage in Skyscanner is preferred.

**Pull request merged**